nntrainer/src/neuralnet.cpp

```diff
@@ -1066,6 +1091,36 @@ int NeuralNetwork::setOptimizer(std::shared_ptr<Optimizer> optimizer) {
   return ML_ERROR_NONE;
 }
 
+void NeuralNetwork::ensureName(std::shared_ptr<Layer> &layer,
```

Comment:
I found this guide helpful https://www.modernescpp.com/index.php/c-core-guidelines-passing-smart-pointer

```suggestion
void NeuralNetwork::ensureName(std::shared_ptr<Layer> layer,
```

Reply:
Done.